### ISSUE TITLE
ReactNativePayments.supportedGateways

### ISSUE THREAD

**Issue by prodrom** [naoufal/react-native-payments#61]
Hi guys,

I imported the library into the project. But when I run it I get such an error. 
What can i do to you ? 

Thanks..

**Comment by Andreyco** [comment]
Have you linked the lib?

**Comment by prodrom** [comment]
I did everything in tutroial and i linked lib. But when i run it i have this (ReactNativePayments.supportedGateways) error.

**Comment by iM-GeeKy** [comment]
I too received the same problem as commented in issue #19.

**Comment by Hulk-Tron** [comment]
I too received this error did search a lot to resolve this issue but my failed getting supportedGateways of undefined error

**Comment by Andreyco** [comment]
This seems to be issue with linking a lib. Could you try again?

**Comment by Hulk-Tron** [comment]
Yes i had succesfully resolve this problem and also understand the problem too. This error is because the payment native sdk is enable to find payment methods you need to install the payment methods manulaly.
Step to resolve:
1. Instal payment cli.
2. Add payment method extension like braintree or stripe using paymentcli.
3. Manually drag or drop payment extension into ur project folder.

**Comment by rmp613** [comment]
@Hulk-Tron Hi,
When you say "drag or drop payment extension" what exactly do you mean? The whole react-native-payments-addon-*** folder?
Cheers

**Comment by Hulk-Tron** [comment]
@rmp613 brother i believe the documentation for react-native-payments is incomplete there are another section which tell you to install payment-cli and install payment extension like stripe or braintree using the payment cli once you install the payment extension using the payment cli manually drop the braintree or stripe extension into framework.
Thanks

**Comment by rmp613** [comment]
@Hulk-Tron Thanks for your reply
I'm still a bit confused sorry. 
I installed payment-cli and the braintree addon. Then I attempted to drag and drop the braintree extension into my frameworks folder and ensured they are linked but I still get the same supported gateways error :(
Would you mind further explaining the steps required?
For example:
After installing payment-cli and the extension using the payment cli I went into the 
`node_modules/react-native-payments-addon-braintree/Carthage/Build/ios`
folder and dragged all of the frameworks into my frameworks folder.
Then I made sure they were all linked to my project in the "Linked Frameworks and Libraries" part of the general tab in xcode.
Then when I rebuilt I had the same error:
[`tid:com.facebook.react.ExceptionsManagerQueue] Unhandled JS Exception: Cannot read property 'supportedGateways' of undefined`

Cheers

**Comment by davekedar** [comment]
@Hulk-Tron can you please help us understand how you have solved this ?

**Comment by lenichols** [comment]
+1
